File: java/src/main/java/vsdk/toolkit/raycasting/photonMap/NormalQuery.java
package vsdk.toolkit.raycasting.photonMap;

import vsdk.toolkit.common.linealAlgebra.Vector3D;

public class NormalQuery {
    public IrrPhoton photon;
    public float[] point;
    public Vector3D normal;
    public float threshold;
    public float maximumDistance;

    public NormalQuery() {
        photon = null;
        point = null;
        normal = new Vector3D();
        threshold = 0.0f;
        maximumDistance = 0.0f;
    }
}

File: typescript/base/src/main/vsdk/toolkit/raycasting/photonMap/NormalQuery.ts
import { Vector3D } from "../../common/linealAlgebra/Vector3D";
import { IrrPhoton } from "./IrrPhoton";

export class NormalQuery {
  public photon: IrrPhoton | null;
  public point: number[] | null;
  public normal: Vector3D;
  public threshold: number;
  public maximumDistance: number;

  public constructor() {
    this.photon = null;
    this.point = null;
    this.normal = new Vector3D();
    this.threshold = 0.0;
    this.maximumDistance = 0.0;
  }
}

